### PR TITLE
[MAGETWO-1556] Export: Unable to Filter Data by Attribute With Input …

### PR DESCRIPTION
**Pull request by php4umagento**
…Type Multiple Select

 - added support for multiselect attribute for both product and customer entity

### Fixed Issues (if relevant)

1. magento-engcom/import-export-improvements#33: [MAGETWO-1556] Export: Unable to Filter Data by Attribute With Input Type Multiple Select

### Manual testing scenarios

1. Add multiselect attribute to customer entity
2. Populate data for that newly created attribute
3. Go to Import/Export and select Customer Main data
4. Observe that multiselect attributes are presented as multiselect form field
5. Data is filtered by selected values

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by dmanners** [comment]
Hey @php4umagento it overall looks good. Now that it is working I would see this as a good opportunity to refactor and tidy the code up a bit. For example the current static tests complain about `The method getAttributeFilterType() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.` https://travis-ci.org/magento-engcom/import-export-improvements/jobs/330107223 I know this is not your fault but it would be great if we could some how sort this issue.

Other than that it looks great.

**Comment by magento-engcom-team** [comment]
@php4umagento thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by dmanners** [comment]
Code wise this is looking good to me @php4umagento I will run this through our internal tools and the QA team and see if anything else comes up.

File: app/code/Magento/ImportExport/Model/Export.php [hunk 1]
```diff
@@ -65,6 +67,17 @@ class Export extends \Magento\ImportExport\Model\AbstractModel
      */
     protected $_exportAdapterFac;
 
+    /**
+     * @var array
+     */
+    public static $backendTypeToFilterMapper = [
```
Comment (dmanners) [inline app/code/Magento/ImportExport/Model/Export.php]: Is there any current need for these to be public?

Comment (php4umagento) [reply app/code/Magento/ImportExport/Model/Export.php]: changed to protected

**Comment by php4umagento** [comment]
@dmanners is there anything else for me here?

**Comment by dmanners** [comment]
@php4umagento just that we do not encourage protected visibility could we make that a change?

**Comment by php4umagento** [comment]
build is running now with "private" scope

**Comment by php4umagento** [comment]
@dmanners done

**Comment by dmanners** [comment]
Thanks @php4umagento I will get this PR into the process loop now and see if we can sort some QA out for this task.

**Comment by magento-engcom-team** [comment]
@php4umagento thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by dmanners** [comment]
Hi @php4umagento during testing the QA team validated the customer section is fixed but would it also be possible to fix the product filter part as well?

```
Steps to reproduce

1) Create product attribute with type multiselect and add to attribute set.
2) Create several product with created attribute.
3) Go to Export and select Products
4) Observe that multiselect attributes are presented as multiselect form field
5) Select multiselect attribute and any option in filter
6) Press "Continue" button
 Expected result
   - Data is filtered by selected option.
 Actual result
   - Exported all products, not looking at the filter.
```

**Comment by dmanners** [comment]
@php4umagento 

```
Preconditions

There is a custom attribute created with input type Multiple Select, 3 options enabled ("Option 1", "Option 2", "Option 3").
There is a customer in the system created with custom attribute value = "Option 1" and second customer in the system created with custom attribute value = "Option 2".
Steps to reproduce

Go to System-> Import/Export-> Export at backend.
Select "Customers" entity type.
In the grid for attribute from precondition select "Option 1" and "Option 2" filter.
Press "Continue" button.
 Expected Result
The file was exported with customers from precondition.

 Actual Result
File was not exported.
The error message appears "There is no data for export".
 The same behavior with products.
```

**Review by dmanners — APPROVED** [review]

**Pull request merged**